### PR TITLE
RPC: Plug getConfirmedSignaturesForAddress2 into bigtable storage

### PR DESCRIPTION
**Pull request by mvines**

### CODE REVIEW

File: ledger-tool/src/bigtable.rs [hunk 1]
```diff
@@ -308,30 +308,39 @@ async fn confirm(signature: &Signature, verbose: bool) -> Result<(), Box<dyn std
 
 pub async fn transaction_history(
```
Comment (mvines) [inline ledger-tool/src/bigtable.rs:309]: The changes in here are actually not directly related :)

Big table chokes if you ask it to return hundreds of thousands of row keys at once, so here a single BigTable query is just being broken up into many smaller queries

File: ledger-tool/src/bigtable.rs [hunk 2]
```diff
@@ -462,7 +471,7 @@ impl BigTableSubCommand for App<'_, '_> {
                                 .value_name("LIMIT")
                                 .validator(is_slot)
                                 .index(2)
-                                .default_value("1000")
+                                .default_value("18446744073709551615")
```
Comment (CriesofCarrots) [inline ledger-tool/src/bigtable.rs:474]: Just noticed: the `about` for this subcommand says: "Show historical transactions affecting the given address, ordered based on the slot in which they were confirmed in from lowest to highest slot"

Looks like it's copied from cli/cluster_query.rs, but needs updating there since #11259

File: core/src/rpc.rs [hunk 3]
```diff
@@ -859,6 +859,25 @@ impl JsonRpcRequestProcessor {
...
+                if let Some(bigtable_ledger_storage) = &self.bigtable_ledger_storage {
+                    let limit = limit - results.len();
+                    let start_after = results.last().map(|x| &x.signature);
+
+                    let mut bigtable_results = self
+                        .runtime_handle
+                        .block_on(
+                            bigtable_ledger_storage.get_confirmed_signatures_for_address(
+                                &address,
+                                start_after,
```
[the remark this answers is not in the document]
Comment (mvines) [reply core/src/rpc.rs]: done

File: core/src/rpc.rs [hunk 4]
```diff
@@ -859,6 +859,25 @@ impl JsonRpcRequestProcessor {
                 )
                 .map_err(|err| Error::invalid_params(format!("{}", err)))?;
 
+            if results.len() < limit {
+                if let Some(bigtable_ledger_storage) = &self.bigtable_ledger_storage {
+                    let limit = limit - results.len();
+                    let start_after = results.last().map(|x| &x.signature);
```
[the remark this answers is not in the document]
Comment (mvines) [reply core/src/rpc.rs]: Ah, thanks . I've reworked that commit to check for empty
